src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo_Metadata.cs

```diff
@@ -26,6 +27,30 @@ namespace Microsoft.CodeAnalysis.FindSymbols
 {
     internal partial class SymbolTreeInfo
     {
+        /// <summary>
+        /// Cache the symbol tree infos for assembly symbols produced from a particular <see
+        /// cref="PortableExecutableReference"/>. Generating symbol trees for metadata can be expensive (in large
+        /// metadata cases).  And it's common for us to have many threads to want to search the same metadata
+        /// simultaneously. As such, we use an AsyncLazy to compute the value that can be shared among all callers.
+        /// <para>
+        /// We store this keyed off of the <see cref="Checksum"/> produced by <see cref="GetMetadataChecksum"/>.  This
+        /// ensures that 
+        /// </para>
+        /// </summary>
+        private static readonly ConditionalWeakTable<PortableExecutableReference, AsyncLazy<SymbolTreeInfo>> s_peReferenceToInfo = new();
+
+        /// <summary>
+        /// Similar to <see cref="s_peReferenceToInfo"/> except that this caches based on metadata id.  The primary
+        /// difference here is that you can have the same MetadataId from two different <see
+        /// cref="PortableExecutableReference"/>s, while having different checksums.  For example, if the aliases of a
+        /// <see cref="PortableExecutableReference"/> are changed (see <see
+        /// cref="PortableExecutableReference.WithAliases(IEnumerable{string})"/>, then it will have a different
+        /// checksum, but same metadata ID.  As such, we can use this table to ensure we only do the expensive
+        /// computation of the <see cref="SymbolTreeInfo"/> once per <see cref="MetadataId"/>, but we may then have to
+        /// make a copy of it with a new <see cref="Checksum"/> if the checksums differ.
+        /// </summary>
+        private static readonly ConditionalWeakTable<MetadataId, AsyncLazy<SymbolTreeInfo>> s_metadataIdToSymbolTreeInfo = new();
```

Comment:
this is the crux of hte change and the thing i missed which was causing the problem.  THere can ne a many->one mapping from PERefs to a particular MetadataId.  But i was keying only off metadata-id, assuming it would be 1:1.

Reply:
Interesting :)